bumble/device.py

```diff
@@ -2230,45 +2230,45 @@ def on_connection(
             self.advertising_own_address_type = None
             self.advertising = False
 
-            # Create and notify of the new connection asynchronously
-            async def new_connection():
-                # Figure out which PHY we're connected with
-                if self.host.supports_command(HCI_LE_READ_PHY_COMMAND):
-                    result = await asyncio.shield(self.send_command(
-                        HCI_LE_Read_PHY_Command(connection_handle=connection_handle),
-                        check_result=True,
-                    ))
-                    phy = ConnectionPHY(
-                        result.return_parameters.tx_phy, result.return_parameters.rx_phy
-                    )
-                else:
-                    phy = ConnectionPHY(HCI_LE_1M_PHY, HCI_LE_1M_PHY)
-
+            if own_address_type in (
+                OwnAddressType.PUBLIC,
+                OwnAddressType.RESOLVABLE_OR_PUBLIC,
+            ):
+                self_address = self.public_address
+            else:
                 self_address = self.random_address
-                if own_address_type in (
-                    OwnAddressType.PUBLIC,
-                    OwnAddressType.RESOLVABLE_OR_PUBLIC,
-                ):
-                    self_address = self.public_address
-
-                # Create a new connection
-                connection = Connection(
-                    self,
-                    connection_handle,
-                    transport,
-                    self_address,
-                    peer_address,
-                    peer_resolvable_address,
-                    role,
-                    connection_parameters,
-                    phy,
-                )
-                self.connections[connection_handle] = connection
 
+            # Create a new connection
+            connection = Connection(
+                self,
+                connection_handle,
+                transport,
+                self_address,
+                peer_address,
+                peer_resolvable_address,
+                role,
+                connection_parameters,
+                ConnectionPHY(HCI_LE_1M_PHY, HCI_LE_1M_PHY),
+            )
+            self.connections[connection_handle] = connection
+
+            if not self.host.supports_command(HCI_LE_READ_PHY_COMMAND):
```

Comment:
nit: it may be more natural to switch the order of the `if` here (`if x: ... else: ...` instead of `if not x: ... else: ...`)